### PR TITLE
Add test for node connection (MAID-1534)

### PR DESCRIPTION
**Pull request by madadam**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/891)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @canndrew (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

File: src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+fn connect() {
+  let (event_sender, event_receiver) = mpsc::channel();
+
+  let nodes_count = 4;
+  let timeout = Duration::from_secs(10);
+
+  let mut nodes = Vec::with_capacity(nodes_count);
+  let mut connection_counts = Vec::with_capacity(nodes_count);
+
+  for i in 0..4 {
```
Comment (afck) [inline src/tests.rs]: Should probably be `0..nodes_count`.

**Comment by afck** [comment]
The indentation is off in some lines: It should always be 4 spaces.

File: src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+use std::time::Duration;
+use maidsafe_utilities::thread::RaiiThreadJoiner;
+
+use event::Event;
+use node::Node;
+
+struct TestNode {
+  // id: usize,
+  _node: Node,
+  _thread_joiner: RaiiThreadJoiner,
```
Comment (afck) [inline src/tests.rs]: I don't think these need leading underscores. They are being used, so they shouldn't cause any "unused" errors.

Comment (madadam) [inline src/tests.rs]: The linter is complaining about unused struct fields. I guess because they are only assigned, but never referenced again.

Comment (afck) [inline src/tests.rs]: Ah, right - and they are needed so that the node isn't dropped and the thread isn't joined.

Comment (madadam) [inline src/tests.rs]: That's right.

**Pull request merged**